rust/lang/security/audit/xss-templates.yaml

```diff
@@ -0,0 +1,24 @@
+rules:
+  - id: cross-site-scripting-templates-detected
+    patterns:
+      - pattern-either:
+          - pattern: |
+              let $VAR = requests::$METHOD(...).unwrap();
+              ...
+              return [...,$VAR,...].join("\n");
+
```

Comment:
Taint mode!

Also think more about how to send HTML back. Commonly we use direct response writers as taint sinks for XSS. I haven't used Rust beyond a few small tests, but what I came up with in the Rocket framework for example was this:

```
#[get("/xss?<param>")]
fn xss(param: String) -> Html<String> {
    Html(format!("Html {}", param))
}
```

In this case the `Html(...)` function could be a taint sink, for example.